pkg/plugins/kafka/consumer.go

```diff
@@ -115,7 +115,7 @@ func (c *segmentConsumer) Close() {
 	// this will also make the loops in the reader goroutines stop
 	err := c.reader.Close()
```

Comment:
It would make sense to return the error here and then return the error in `Teardown`. We could sneak it into a PR once we touch this code again!

Reply:
👍 Sounds good!